src/raspi/component/can/can_bus.py

```diff
@@ -1,64 +1,64 @@
 import can
-import cantools
-import math
-import time
 import logging
+from typing import Callable
+from dataclasses import dataclass
 
+from state_management import (
+    create_generic_context,
+    device,
+    device_action,
+    device_parser,
+)
+
+@device
+@dataclass(slots = True)
 class CanBus:
     bus: can.Bus
-    db: object
+    listeners: dict
+    notifier : can.Notifier
 
     def __init__(self,channel = "can0",interface = "socketcan",config_context = None, ignore_config = False, **kwargs):
-        self.db = cantools.database.load_file("src/raspi/odrive-cansimple.dbc")
         self.bus = can.Bus(channel,interface,config_context,ignore_config,**kwargs)
-    
-
-# TODO: Move MotorController into it's own file.
-class MotorController:
-    axis_id: int
-    can: CanBus
+        self.listeners = {}
+        self.notifier = can.Notifier(self.bus, [lambda msg: read_message(self, msg)])
```

Comment:
```py
DEFAULT_BUS_CONFIG = {"channel": "can0", "interface": "socketcan", "config_context": None, ignore_config: False}

class CanBus:
    busConfig: dict = DEFAULT_BUS_CONFIG
    bus: can.Bus = None
    notifier: can.Notifier = None
    listeners = field(default_factory=dict)

    def __post_init__(self):
        self.busConfig = {**DEFAULT_BUS_CONFIG, **self.busConfig}
        self.bus = self.bus if self.bus else can.Bus(**self.busConfig) 
        self.notifier = self.notifier if self.notifier else can.Notifier(self.bus, (lambda msg: read_message(self, msg),))
```
I'd recommend this implementation for several reasons:
1. While we probably will not have more than one CanBus object, there will be errors of multiple CanBus objects all sharing the same listeners dictionary object, so `field(default_factory=dict)` will fix this bug and is a good practice to begin with.
2. Debugging with __repr__. You will be able to see all of the config params when converted to a string
3. Able to dependency inject during testing